### DISCUSSION TITLE
Long time PWM-Meassuring

### DISCUSSION THREAD

**Discussion by dettkima** [OpenHantek/OpenHantek6022#410] [Q&A] [answered]
Hey there, i am using a Hantek 6022BE for my Bachelor thesis. I need to monitor a 1kHz PWM-Signal over 10 hours, and i need the duty cycle and the maximum voltage of each period. Is it somehow possible to configrate the data export to save this as csv? I saw the duty-cycle in the top left corner, so i hope that there might be a way to export these Values, so I wont have to store all the 'useless' data to analyse it afterwards with python.

**Comment by jonw0224** [comment, marked as answer]
dettkima,

The best way to do this is to use the Hantek6022API, which utilizes Python to make captures from the scope. It's easy to use. You'd write your own script to do what you want to do. Unfortunately, you'll need to analyze the waveform of each capture to calculate the duty cycle and maximum voltage. It would be a good learning opportunity for your thesis. Here's a link to the project on GitHub: 

[https://github.com/Ho-Ro/Hantek6022API](https://github.com/Ho-Ro/Hantek6022API)

Good luck!

Jonathan

**Reply by dettkima** [reply]
Hi jonw0224,
thank you very much for your awnser. I also saw the API and hope it will work for me. As i am quite new to programming I hoped that it could work with the OpenHantek6022, but seems like I have to improove my programming skills and go through it with the API. Its quite good to know that you think that it should work with the API, so I know I am on the right way and just have to find out how.

**Comment by Ho-Ro** [comment]
> I need to monitor a 1kHz PWM-Signal over 10 hours, and i need the duty cycle and the maximum voltage of each period.

Not possible with OH6022 b/c the duty cycle display is only valid for the **triggered** edge and the program skips data during search for the next trigger edge (triggering is done completely in SW due to the limited HW of the scope). But you can use the (almost) infinite [capturing tool](https://github.com/Ho-Ro/Hantek6022API/blob/main/examples/capture_6022.py) from https://github.com/Ho-Ro/Hantek6022API

EDIT: I typed too slow, thx Jonathan :)

**Reply by jonw0224** [reply]
Great minds think alike!

 On the duty cycle you could do something like calculate an average of the captured voltage values and compare to the average, keeping up with the percentage that are above or below the average. 

Getting the maximum voltage would simply involve finding the largest value. That gets a little trickier if you don't want to count voltage spikes if they're present in the waveform, but it could be something like the maximum average value of 4-5 samples to remove spikes in the waveform.

**Reply by dettkima** [reply]
Thank you very much for your awnser. I am a newbie in Programming and using open source stuff and it is really impressive how fast I got an awnser and how well maintained and documentated it is (I had no clue what i am doing and still brought it to life) keep going with your great work!
